src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/tables/Client.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.tables;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.foods.Food;
+import com.epam.training.designpatterns.fastfoodrestaurant.foods.Menu;
+import com.epam.training.designpatterns.fastfoodrestaurant.staff.Server;
+
+import java.util.Random;
+import java.util.concurrent.Semaphore;
+
+/**
+ * Knows the server, and the menu
+ */
+public class Client implements Runnable{
+    private static int clientCount;
+    private final Semaphore sema;
+
+    private double happiness = 100;
+    private final int clientNumber;
+    private Food food;
+    private final int SIMULATION_SPEED;
+    private final Server server;
+
+    public Client(Server server, Semaphore sema, int simulation_speed){
+        SIMULATION_SPEED = simulation_speed;
+        this.clientNumber = clientCount++;
```

Comment:
Ezt a `ClientFactory` beállíthatja, amikor új vedéget gyárt és ő fogja számlálni, hogy mennyit gyártott, mi az aktuális sorszám.
Ebben az esetben nem gyártunk több szálról, de biztonságosabb ez a megoldás:
```
private AtomicInteger counter;

public int getNextUniqueIndex() {
    return counter.getAndIncrement();
}
```
forrás: https://stackoverflow.com/questions/4818699/practical-uses-for-atomicinteger